block_synchronizer/deploy_acquisition: add unit tests

alsrdn opened the pull request:
Unit tests for the deploy acquisition logic as part of https://github.com/casper-network/casper-node/issues/3437.

alsrdn committed:
block_synchronizer/deploy_acquisition: add unit tests

Signed-off-by: Alexandru Sardan <alexandru@casperlabs.io>

rafal-ch commented on node/src/components/block_synchronizer/deploy_acquisition/tests.rs:
```diff
@@ -0,0 +1,176 @@
+
+    assert_matches!(
+        deploy_acquisition.apply_approvals_hashes(&approvals_hashes),
+        Err(Error::AcquisitionByIdNotPossible)
```
Can you add a test covering the [EncounteredNonVacantDeployState](https://github.com/casper-network/casper-node/blob/dev/node/src/components/block_synchronizer/deploy_acquisition.rs#L15) error variant?

alsrdn replied:
added here: https://github.com/casper-network/casper-node/pull/3690/commits/c5117c8e0139028ad0e64f020d7e8e06b04143cc

rafal-ch reviewed (approved):
LGTM :+1: 
with one minor suggestion :)

Fraser999 reviewed (approved):
Approved assuming Rafal's ask is satisfied.

alsrdn committed:
block_synchronizer/deploy_acquisition: add another test

Add test for applying approvals hashes when deploy was already acquired
by hash.

Signed-off-by: Alexandru Sardan <alexandru@casperlabs.io>

alsrdn commented:
bors try

alsrdn commented:
bors r+
